JetStreamCompose/jetstream/src/main/java/com/google/jetstream/presentation/screens/videoPlayer/components/VideoPlayerState.kt

```diff
@@ -37,18 +43,42 @@ import kotlinx.coroutines.flow.debounce
 class VideoPlayerState(
     @IntRange(from = 0)
     private val hideSeconds: Int,
-    val playPauseButtonState: PlayPauseButtonState,
+    private val playPauseButtonState: PlayPauseButtonState,
```

Comment:
Instead of wrapping all button states into a parent state, you can also consider passing the player instance down the UI hierarchy and letting each control create its own state. The sample in the media3 library shows an example of that.

Reply:
Thanks for the suggestion. Updated the PR that creates some components to hold the state objects as suggested.